### PR TITLE
addtion of analyze button and fix for dashboard modal

### PR DESCRIPTION
**Pull request by samcaspus**
## Changes

*Please describe.*  
Tried to fix issue https://github.com/PostHog/posthog/issues/2626

image

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

**Review by paolodamico — CHANGES_REQUESTED** [review]
Hey @samcaspus! Thanks for taking the time to add this contribution. Some feedback,

1. I think it'll be best to keep consistency across the app and use the same element we use for feature flags (see screenshot below).

1. Might be worth adding the element directly to the table view (on the main list), to increase discoverability and keep consistency with feature flags. We can iterate later based on usage.

File: frontend/src/scenes/events/EventDetails.js [hunk 1]
```diff
@@ -35,6 +37,15 @@ export function EventDetails({ event }) {
             >
                 Create action from event
             </Button>
+
+            <Link
```
[the remark this answers is not in the document]
Comment (samcaspus) [reply frontend/src/scenes/events/EventDetails.js]: hey not sure if i did this right made an update and pushed 😅

**Comment by samcaspus** [comment]
added a minor fix for (https://github.com/PostHog/posthog/issues/2632) 

image

**Review by paolodamico — CHANGES_REQUESTED** [review]
Hey @samcampus,

- Good catch on the fix for the save dashboards, I have cherry picked your commit to another PR (#2673) and it's good to go.

As for the event insights, some feedback.
- In terms of UI, this is what we would expect to keep consistency with the rest of the app,


here's what the code would look like (based on the way you generate the insights URL) which would go on `EventsTable.js`  around line 160,
```typescript
{
    title: 'Usage',
    key: 'usage',
    render: function render({ event }) {
        const slicedEvent = event.event[0] == '$' ? event.event.slice(1) : event.event
        return (
            <Link
                to={`/insights?insight=TRENDS&interval=day&display=ActionsLineGraph&events=%5B%7B%22id%22%3A%22%24${slicedEvent}%22%2C%22name%22%3A%22%24pageview%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%7D%5D&properties=#backTo=Events&backToURL=${window.location.pathname}`}
                data-attr="events-table-usage"
            >
                Insights <ExportOutlined />
            </Link>
        )
    },
}
```
 
- The other issue with this is that we're not actually correctly generating the insights URL. The URL at least does not work for `$autocapture` events or custom events.

**Comment by samcaspus** [comment]
Trying to figure out a way to incorporate $autocapture events or custom events.

**Comment by samcaspus** [comment]
@paolodamico could you please help me out here i believe its redirecting to $autocapture 
and i am not able to see any event without $ getting selected. 
if any changes do let me know would be happy to modify the code.

Thanks

**Review by paolodamico — CHANGES_REQUESTED** [review]
Hey @samcaspus, thanks for taking the time to keep working on this. Comments below.

1. Apologies, looks like I forgot to attach the screenshot, however you nailed it, almost exactly right. The only bug with the usage column is that the date group gets cut off at the last column (see screenshot), should be fairly simple to fix.
    image

1. Related to opening the insights page, I think we can separate the types of events into three (and handle each separately), `$pageview`s, `$autocapture` and custom events.
    - For `$pageview` events you'd want to send `$pageview` as the event, but matching the URL of the event specifically. Example: ```/insights?insight=TRENDS&interval=day&display=ActionsLineGraph&events=%5B%7B%22id%22%3A%22%24pageview%22%2C%22name%22%3A%22%24pageview%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%2C%22properties%22%3A%5B%7B%22key%22%3A%22%24current_url%22%2C%22value%22%3A%22https%3A%2F%2Fapp.posthog.com%2Fevents%22%2C%22operator%22%3A%22exact%22%2C%22type%22%3A%22event%22%7D%5D%7D%5D&properties=%5B%5D&actions=%5B%5D&new_entity=%5B%5D```
    - For custom events, it's fairly straightforward, you just need to filter for that specific custom event. Example: ```/insights?insight=TRENDS&interval=day&display=ActionsLineGraph&actions=%5B%5D&events=%5B%7B%22id%22%3A%22Order%20Completed%22%2C%22name%22%3A%22Order%20Completed%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%7D%5D&properties=%5B%5D&new_entity=%5B%5D```
    - `$autocapture` is tricky, because there's no good way of measuring direct `$autocapture` events on an insights graph, I think the best thing to do here is instead of adding an insights button, to add a create action button (be sure to use the actions icon).

Let me know if you need any help or get stuck somewhere.

**Comment by samcaspus** [comment]
hey @paolodamico would be great if you could review it once more and let me know if further change is required 😄 

Thanks

**Comment by paolodamico** [comment]
hey @samcaspus! apologies for the delay here, I've been swamped. I'll review again today and get back to you, thanks for the nudge!

**Comment by paolodamico** [comment]
Closing in favor of #2748 to add some final tweaks. Thanks for the contribution!

**Pull request closed without merging**